src/pgwire/src/protocol.rs

```diff
@@ -280,8 +281,17 @@ where
 
         self.coord_client.reset_canceled();
 
+        // TODO(guswynn): add spans to all message types. Currently we don't
+        // to avoid useless spans
```

Comment:
Wait is this a `TODO` or not then? It seems unsure about whether we should or shouldn't.

Reply:
This `TODO` is real, we only use actually make new spans for `Query` and `Execute` right now

Reply:
In the future, if we want to trace all message types (its unclear to me if we do), then we just lift the span creation to the top of the function and `.instrument` the entire body

Reply:
Right right, I just meant comment currently reads (to me) as "TODO: do FOO. Currently we don't do FOO because FOO is useless." :D I left you some alternate wording for consideration below!